p2p/src/utils/mux.rs

```diff
@@ -186,15 +212,6 @@ impl<P: PublicKey> Receiver for SubReceiver<P> {
     }
 }
 
-impl<P: PublicKey> Drop for SubReceiver<P> {
```

Comment:
May be worthwhile to either add an explicit `Deregister` or re-add with a `try_send` here (or just make the control channel unbounded).

Reply:
(I think `unbounded` doesn't require await but may be wrong there...)